com.greyrook.siptop.json

```diff
@@ -0,0 +1,78 @@
+{
+  "app-id": "com.greyrook.siptop",
+  "base": "org.electronjs.Electron2.BaseApp",
+  "base-version": "19.08",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "19.08",
+  "sdk": "org.freedesktop.Sdk",
+  "command": "siptop.sh",
+  "separate-locales": false,
+  "finish-args": [
+    "--share=ipc",
+    "--socket=x11",
+    "--socket=pulseaudio",
+    "--share=network",
+    "--device=all",
+    "--filesystem=xdg-documents:ro",
+    "--filesystem=xdg-pictures:ro",
+    "--filesystem=xdg-music:ro",
+    "--filesystem=xdg-videos:ro",
+    "--filesystem=xdg-download",
+    "--filesystem=xdg-run/dconf",
+    "--filesystem=~/.config/dconf:ro",
+    "--env=DCONF_USER_CONFIG_DIR=.config/dconf",
+    "--talk-name=ca.desrt.dconf",
+    "--talk-name=org.freedesktop.Notifications",
+    "--env=XDG_CURRENT_DESKTOP=Unity",
+    "--talk-name=org.kde.StatusNotifierWatcher"
+  ],
+  "modules": [
+    {
+      "name": "siptop",
+      "buildsystem": "simple",
+      "build-commands": [
+        "cp -r siptop-0.1.0/* ${FLATPAK_DEST}/bin/",
+        "install siptop.sh ${FLATPAK_DEST}/bin/siptop.sh",
+        "install -Dm644 com.greyrook.siptop.metainfo.xml ${FLATPAK_DEST}/share/appdata/${FLATPAK_ID}.metainfo.xml",
+        "install -Dm644 com.greyrook.siptop.icon.png ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png",
+        "install -Dm644 com.greyrook.siptop.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop",
+        "desktop-file-edit --set-key=\"Icon\" --set-value=\"com.greyrook.siptop\" ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop",
+        "desktop-file-edit --set-key=\"Exec\" --set-value=\"siptop.sh %U\" ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop",
+        "desktop-file-edit --set-key=\"StartupWMClass\" --set-value=\"siptop\" ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop",
+        "desktop-file-edit --set-key=\"X-Flatpak-RenamedFrom\" --set-value=\"siptop.desktop;\" ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop",
+        "install -Dm755 /usr/bin/ar -t ${FLATPAK_DEST}/bin",
+        "install -Dm755 /usr/lib/$(gcc -print-multiarch)/libbfd-*.so -t ${FLATPAK_DEST}/lib"
+      ],
+      "sources": [
+        {
+          "type": "script",
+          "dest-filename": "siptop.sh",
+          "commands": [
+            "exec env TMPDIR=$XDG_CACHE_HOME /app/bin/siptop --no-sandbox\"$@\""
```

Comment:
Use zypak and zypak-wrapper instead of --no-sandbox.

Reply:
Any updates on this?

Reply:
I took a look at [irccloud](https://github.com/flathub/com.irccloud.desktop/search?q=zypak&unscoped_q=zypak) but did not get where yzpak gets installed. So I assumed it is part of the base image but it seems it is not.

Another pointer would be welcome, while proficient with linux and packages we both are still new to building flatpacks :)

Thanks for your patience.

Reply:
zypak is part of  `org.electronjs.Electron2.BaseApp` https://github.com/flathub/org.electronjs.Electron2.BaseApp/blob/branch/19.08/org.electronjs.Electron2.BaseApp.yml#L81

Reply:
@bilelmoussaoui thank you for your quick feedback!

we use it as base:
```
  "base": "org.electronjs.Electron2.BaseApp",
  "base-version": "19.08",
```

and I am getting:
```
exec: zypak-wrapper: not found
```

Reply:
I published my current state here https://github.com/FlorianLudwig/flathub/tree/siptop

Reply:
It builds and runs fine here after applying this patch
```diff
diff --git a/com.greyrook.siptop.json b/com.greyrook.siptop.json
index 4ff2c07..2ad2b4d 100644
--- a/com.greyrook.siptop.json
+++ b/com.greyrook.siptop.json
@@ -39,8 +39,9 @@
         {
           "type": "script",
           "dest-filename": "siptop.sh",
-          "commands": [
-            "exec env TMPDIR=$XDG_CACHE_HOME /app/bin/siptop --no-sandbox\"$@\""
+          "commands": [            
+            "export TMPDIR=\"$XDG_RUNTIME_DIR/app/$FLATPAK_ID\"",
+            "exec zypak-wrapper /app/bin/siptop \"$@\""
           ]
         },
         {
@@ -59,7 +60,7 @@
         },
         {
           "type": "archive",
-          "url": "https://github.com/GreyRook/siptop/releases/download/v0.1.0/siptop-0.1.0.tar.gz",
+          "url": "https://github.com/GreyRook/siptop/releases/download/v0.1.0/siptop-0.1.0.x86_64.tar.gz",
           "sha256": "372855af895f09b437d92580e5e8cf80e0ec3603512a8a049d29b8ec15242180",
           "only-arches": ["x86_64"],
           "strip-components": 0

```

Reply:
It is building for me to but not running.
```
$ flatpak run com.greyrook.siptop
/app/bin/siptop.sh: line 3: exec: zypak-wrapper: not found
```

Reply:
i did a clean build like this:
```
flatpak remove -y com.greyrook.siptop

cd tmp/
git clone https://github.com/FlorianLudwig/flathub.git
cd flathub/
git checkout siptop
flatpak-builder build com.greyrook.siptop.json --install --user --force-clean
flatpak run com.greyrook.siptop
``

Reply:
What's the output of `flatpak info org.electronjs.Electron2.BaseApp` ?

Reply:
```
$ flatpak info org.electronjs.Electron2.BaseApp
          ID: org.electronjs.Electron2.BaseApp
         Ref: app/org.electronjs.Electron2.BaseApp/x86_64/19.08
        Arch: x86_64
      Branch: 19.08
      Origin: flathub
  Collection: org.flathub.Stable
Installation: user
   Installed: 1,2 MB
     Runtime: org.freedesktop.Platform/x86_64/19.08
         Sdk: org.freedesktop.Sdk/x86_64/19.08

      Commit: f02eee5d0e2b42a401f9e65b4f7157d7d63b2995eff6c7654942bbe1fea0f920
     Subject: Update shared-modules (f42e13f1)
        Date: 2019-09-25 16:04:24 +0000
```

Reply:
I also tried:
```
 flatpak install --reinstall org.electronjs.Electron2.BaseApp
```
Now I am getting:
```
$ flatpak info org.electronjs.Electron2.BaseApp
          ID: org.electronjs.Electron2.BaseApp
         Ref: app/org.electronjs.Electron2.BaseApp/x86_64/19.08
        Arch: x86_64
      Branch: 19.08
      Origin: flathub
  Collection: org.flathub.Stable
Installation: user
   Installed: 1,6 MB
     Runtime: org.freedesktop.Platform/x86_64/19.08
         Sdk: org.freedesktop.Sdk/x86_64/19.08

      Commit: 88b7a27ba7bc18a2bb9f1bbf7256f6f995060cb9cda80a203cd28455f5f07eb6
      Parent: a564d357f957143fe27c6f25093399b8a7dbe800fbbece5f2fb1aaacf16c3be8
     Subject: Update shared-modules (3366c618)
        Date: 2020-03-20 01:33:33 +0000
```

Reply:
Interesting: After the `--reinstall` starting siptop did still fail.  Then I uninstalled and rebuild it and now it is running.

I did assume that updating `org.electronjs.Electron2.BaseApp` would change the runtime or are flatpaks build on a specific/excat commit/release?

Reply:
You had a very old release indeed. You can just run `flatpak update org.electronjs.Electron2.BaseApp` next time. You might need to add `--user` if you didn't install it globally.

Reply:
merge request updated, does no use `zypak-wrapper`.

Do we need to specify a minimum dependency? Or does the flatpak install with the exact version I build with?

Reply:
The BaseApp is different from a runitme, it's used during build and not at runtime. So yeah, the users will get your app with whatever required to run it.

Reply:
Thanks very much for helping me understand :) Makes a sense now!